hack/test-e2e.sh

```diff
@@ -24,22 +24,27 @@ if [ "${REMOTE_CLUSTER:-false}" = false ] ; then
   sudo sysctl -w vm.max_map_count=262144 ||:
 fi
 
-if oc get project openshift-logging > /dev/null 2>&1 ; then
-  echo using existing project openshift-logging
+TEST_NAMESPACE="${TEST_NAMESPACE:-e2e-test}"
+if oc get project ${TEST_NAMESPACE} > /dev/null 2>&1 ; then
+  echo using existing project ${TEST_NAMESPACE}
 else
-  oc create namespace openshift-logging
+  oc create namespace ${TEST_NAMESPACE}
 fi
 
-oc create -n openshift-logging -f \
+sed -i "s/namespace: openshift-logging/namespace: ${TEST_NAMESPACE}/g" ${manifest}
+
+oc create -n ${TEST_NAMESPACE} -f \
 https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/prometheusrule.crd.yaml || :
-oc create -n openshift-logging -f \
+oc create -n ${TEST_NAMESPACE} -f \
 https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/servicemonitor.crd.yaml || :
 
-TEST_NAMESPACE=openshift-logging go test ./test/e2e/... \
+TEST_NAMESPACE=${TEST_NAMESPACE} go test ./test/e2e/... \
   -root=$(pwd) \
   -kubeconfig=${KUBECONFIG} \
   -globalMan manifests/04-crd.yaml \
   -namespacedMan ${manifest} \
   -v \
   -parallel=1 \
   -singleNamespace
```

Comment:
I think if we update this and remove `TEST_NAMESPACE` then `go test` will randomly create/destroy a namespace. we would need to update the manifest files to not have a `namespace` field on them so that they can be created in the randomly created namespace.

Reply:
We don't know the namespace name what will be created by SDK.

Namespace has to be specified otherwise `ClusterRoleBinding` will fail on creation.

We could fix it by changing `ClusterRole` to `Role` and `ClusterRoleBinding` to `RoleBinding` but it also requires removing `nonResourceURLs` from role.

Reply:
ack, good point, i forgot about the cluster role binding case

Reply:
So is this PR fine? We could try to create a random namespace - e.g. get some random string in shell script.

Reply:
can we copy manifest dir onto a tmp filesystem and replace the namespace there just for the tests?

Reply:
@josefkarasek  this PR already does that. 

I will do a simple change to include a random string in the namespace. I need to get this PR merged today or tomorrow. On Friday I am leaving on PTO for 2 weeks.

Reply:
I have added `$RANDOM` to the namespace. 

I am happy to experiment with removing  ` -singleNamespace`, but that will need more changes. I think we could use this for now and I can work more on it once I am back.

Reply:
I think if we're going to generate a namespace to test in, we should keep `-singleNamespace`

Reply:
I mean if we later decide to rely purely on SDK to create namespace.

Reply:
I think since we will be publishing the manifest files, we should use them when testing.